src/main/java/teammates/ui/webapi/CreateAccountRequestAction.java

```diff
@@ -1,56 +1,209 @@
+
+import org.apache.http.HttpStatus;
+
 import teammates.common.datatransfer.attributes.AccountRequestAttributes;
 import teammates.common.exception.EntityAlreadyExistsException;
+import teammates.common.exception.EntityDoesNotExistException;
 import teammates.common.exception.InvalidParametersException;
+import teammates.common.util.Const;
 import teammates.common.util.EmailWrapper;
+import teammates.common.util.FieldValidator;
+import teammates.common.util.Logger;
+import teammates.common.util.StringHelper;
+import teammates.ui.output.AccountRequestCreateErrorResults;
 import teammates.ui.output.JoinLinkData;
-import teammates.ui.request.AccountCreateRequest;
+import teammates.ui.request.AccountRequestCreateIntent;
+import teammates.ui.request.AccountRequestCreateRequest;
+import teammates.ui.request.AccountRequestType;
 import teammates.ui.request.InvalidHttpRequestBodyException;
 
 /**
  * Creates a new account request.
  */
-class CreateAccountRequestAction extends AdminOnlyAction {
+class CreateAccountRequestAction extends Action {
+
+    private static final Logger log = Logger.getLogger();
+
+    @Override
+    AuthType getMinAuthLevel() {
+        return AuthType.PUBLIC;
+    }
+
+    @Override
+    void checkSpecificAccessControl() throws UnauthorizedAccessException {
+        AccountRequestCreateIntent intent =
+                AccountRequestCreateIntent.valueOf(getNonNullRequestParamValue(Const.ParamsNames.INTENT));
+
+        switch (intent) {
+        case ADMIN_CREATE:
+            if (userInfo == null || !userInfo.isAdmin) {
+                throw new UnauthorizedAccessException("Admin privilege is required to access this resource.");
+            }
+            break;
+        case PUBLIC_CREATE:
+            break;
+        default:
+            throw new InvalidHttpParameterException("Unknown intent " + intent);
+        }
+    }
 
     @Override
     public JsonResult execute() throws InvalidHttpRequestBodyException, InvalidOperationException {
-        AccountCreateRequest createRequest = getAndValidateRequestBody(AccountCreateRequest.class);
+        AccountRequestCreateIntent intent =
+                AccountRequestCreateIntent.valueOf(getNonNullRequestParamValue(Const.ParamsNames.INTENT));
+
+        if (!intent.equals(AccountRequestCreateIntent.ADMIN_CREATE)) {
+            String userCaptchaResponse = getRequestParamValue(Const.ParamsNames.USER_CAPTCHA_RESPONSE);
+            if (userCaptchaResponse == null || !recaptchaVerifier.isVerificationSuccessful(userCaptchaResponse)) {
+                throw new InvalidHttpParameterException("Please check the \"I'm not a robot\" box before submission.");
+            }
+        }
+
+        String type = getRequestParamValue(Const.ParamsNames.ACCOUNT_REQUEST_TYPE);
+        if (type == null || AccountRequestType.valueOf(type) != AccountRequestType.INSTRUCTOR_ACCOUNT) {
+            throw new InvalidHttpParameterException("Only instructor accounts can be created.");
+        }
```

Comment:
This should be checked on the frontend.

Reply:
Backend as well according to @wkurniawan07

Reply:
What? This is not required. This is not a field used in the backend. If someone ticks the "I am a student" box, they should not be allowed to submit the form at all.